core/src/test/java/edu/wpi/grip/core/operations/network/networktables/MockNetworkTable.java

```diff
@@ -0,0 +1,506 @@
+package edu.wpi.grip.core.operations.network.networktables;
+
+import edu.wpi.first.wpilibj.tables.ITable;
+import edu.wpi.first.wpilibj.tables.ITableListener;
+import edu.wpi.first.wpilibj.tables.TableKeyNotDefinedException;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * NetworkTables implementation for tests that doesn't rely on the networking stack.
+ */
+@SuppressWarnings({"OverloadMethodsDeclarationOrder", "PMD"})
+public class MockNetworkTable implements ITable {
+
+  private final Map<String, Object> entries = new HashMap<>();
+  private final Map<String, ITable> subTables = new HashMap<>();
+  private final Map<String, List<ITableListener>> listeners = new HashMap<>();
+  private static final List<Class> SUPPORTED_TYPES = Arrays.asList(
+      byte[].class,
+      String.class,
+      String[].class,
+      Boolean.class,
+      Boolean[].class,
+      boolean[].class,
+      Double.class,
+      Double[].class,
+      double[].class
+  );
+
+  private static final MockNetworkTable ROOT = new MockNetworkTable("/");
```

Comment:
So, for each test we should have a fresh root?
I don't want tests to have static state that can pollute each other.